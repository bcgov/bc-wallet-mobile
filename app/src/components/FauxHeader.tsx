import { ButtonLocation, HeaderButton, testIdWithKey, useTheme } from '@hyperledger/aries-bifold-core'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { View, Text, StyleSheet, Platform } from 'react-native'

interface FauxHeaderProps {
  title: string
  onBackPressed: () => void
}

const FauxHeader: React.FC<FauxHeaderProps> = ({ title, onBackPressed }) => {
  const { ColorPallet, TextTheme } = useTheme()
  const { t } = useTranslation()
  const styles = StyleSheet.create({
    header: {
      backgroundColor: ColorPallet.brand.primary,
      elevation: 0,
      shadowOffset: { width: 0, height: 6 },
      shadowRadius: 6,
      shadowColor: ColorPallet.grayscale.black,
      shadowOpacity: 0.15,
      borderBottomWidth: 0,
      flex: 1,
      flexDirection: 'row',
      alignItems: 'stretch',
      minHeight: Platform.OS === 'ios' ? 44 : 56,
    },
    left: {
      justifyContent: 'center',
      alignItems: 'flex-start',
    },
    titleContainer: {
      marginHorizontal: 16,
      justifyContent: 'center',
      alignItems: 'center',
      flexShrink: 1,
      maxWidth: '68%',
      width: '100%',
    },
    title: {
      ...TextTheme.headerTitle,
      textAlign: 'center',
    },
    right: {
      justifyContent: 'center',
      alignItems: 'flex-end',
    },
  })

  return (
    <View style={styles.header}>
      <View style={styles.left}>
        <HeaderButton
          buttonLocation={ButtonLocation.Left}
          accessibilityLabel={t('Global.Back')}
          testID={testIdWithKey('BackButton')}
          onPress={onBackPressed}
          icon="chevron-left"
        />
      </View>
      <View style={styles.titleContainer}>
        <Text numberOfLines={1} ellipsizeMode="tail" style={styles.title}>
          {title}
        </Text>
      </View>
      <View style={styles.right}></View>
    </View>
  )
}

export default FauxHeader
