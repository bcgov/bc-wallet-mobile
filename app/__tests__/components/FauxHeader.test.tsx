import { render } from '@testing-library/react-native'
import React from 'react'

import FauxHeader from '../../src/components/FauxHeader'

describe('FauxHeader Component', () => {
  test('renders correctly', () => {
    const tree = render(<FauxHeader title={'test'} onBackPressed={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })
})
