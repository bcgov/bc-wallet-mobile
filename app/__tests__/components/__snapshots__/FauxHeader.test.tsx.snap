// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FauxHeader Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#42803E",
      "borderBottomWidth": 0,
      "elevation": 0,
      "flex": 1,
      "flexDirection": "row",
      "minHeight": 44,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityLabel="Global.Back"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="com.ariesbifold:id/BackButton"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 15,
            "marginRight": 0,
            "minHeight": 26,
            "minWidth": 26,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 26,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          Û∞ÖÅ
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexShrink": 1,
        "justifyContent": "center",
        "marginHorizontal": 16,
        "maxWidth": "68%",
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 28,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "center",
      }
    }
  />
</View>
`;
